File: Editorial/2553_Separate_the_Digits_in_an_Array/Go/1_Approach_Simulation.go
func separateDigits(nums []int) []int {
	res := []int{}
	for _, num := range nums {
		x := num
		tmp := []int{}
		for x > 0 {
			tmp = append(tmp, x%10)
			x /= 10
		}
		for i := len(tmp) - 1; i >= 0; i-- {
			res = append(res, tmp[i])
		}
	}
	return res
}

File: Editorial/2553_Separate_the_Digits_in_an_Array/JavaScript/1_Approach_Simulation.js
var separateDigits = function (nums) {
    const res = [];
    for (let num of nums) {
        let x = num;
        const tmp = [];
        while (x > 0) {
            tmp.push(x % 10);
            x = Math.floor(x / 10);
        }
        for (let i = tmp.length - 1; i >= 0; i--) {
            res.push(tmp[i]);
        }
    }
    return res;
};
